src/Symfony/Component/Mime/Email.php

```diff
@@ -412,7 +420,11 @@ public function getBody(): AbstractPart
             return $body;
         }
 
-        return $this->generateBody();
+        if (null === $this->cachedBody) {
```

Comment:
Maybe we can move the check to the beginning of the `generateBody()` just to be safe (right now, it's called only once and private, but I would want to avoid any issues if we changed that)?